src/app/addmarket/page.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import React, { useState, useEffect, ChangeEvent } from 'react';
+import { Layout, Button, message } from 'antd';
+import axios from 'axios';
+import RootLayout from '../layout';
+import { useNavigate } from 'react-router-dom'; // 使用 useNavigate 钩子
+
+// const { Header, Footer, Content } = Layout;
+
+interface Market {
+  exchange: string;
+  name: string;
+  url: string;
+}
+
+interface FormData {
+  exchange: string;
+  key: string;
+  secret: string;
+  desc: string;
+}
+
+const AddApiKeyForm: React.FC = () => {
+  const navigate = useNavigate(); 
+  const [exchangeList, setExchangeList] = useState<Market[]>([]);
+  const [formData, setFormData] = useState<FormData>({
+    exchange: '',
+    key: '',
+    secret: '',
+    desc: '',
+  });
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  useEffect(() => {
+    queryMarketList();
+  }, []);
+
+  const queryMarketList = async () => {
+    // Need to be completed.
+  };
+
+  const handleSelectExchange = ( ) => {
+    // Need to be completed.
+  };
+
+  const handleInputChange = ( ) => {
+    // Need to be completed.
+  };
+
+  const handleAddMarket = async () => {
+    // Need to be completed.
+  };
+
+  const goBack = () => {
+    navigate(-1); 
+  };
+
+  return (
+    <RootLayout>
+      <div className="container mx-auto" style={{ marginTop: '70px', maxWidth: '1200px' }}>
+        <section className="section">
+          <div className="box">
+            <div className="header">
+              <p className="is-size-6 is-pulled-left" style={{ marginRight: '10px' }}>New Exchange API Key</p>
+              <a onClick={goBack} className="is-pulled-right">Go Back</a>
+            </div>
+            <div className="field">
+              <label className="label">Select Exchange</label>
+              <div className="control">
+                <ul className="choice is-clearfix">
+                  {exchangeList.map((item) => (
+                    <li
+                      key={item.exchange}
+                      className={`exchange-item ${formData.exchange === item.exchange ? 'active' : ''}`}
+                      // onClick={() => handleSelectExchange(item.exchange)}
```

Comment:
why not using this onClick function?

Reply:
Because it depends on other unfinished parts, otherwise it will report an error.